baremetal/install.sh

```diff
@@ -0,0 +1,199 @@
+    fi
+    if [[ ! -f "$SCRIPT_DIR/manifests/trusted-compute.yaml" ]]; then
+        print_error "trusted-compute.yaml not found in manifests directory"
+        exit 1
+    fi
+}
+
+# Function to create target directories
+create_target_dirs() {
+    print_status "Creating target directories if they don't exist..."
+    [[ ! -d "$K3S_CHARTS_DIR" ]] && mkdir -p "$K3S_CHARTS_DIR" && print_status "Created charts directory: $K3S_CHARTS_DIR" || print_status "Charts directory already exists: $K3S_CHARTS_DIR"
+    [[ ! -d "$K3S_MANIFESTS_DIR" ]] && mkdir -p "$K3S_MANIFESTS_DIR" && print_status "Created manifests directory: $K3S_MANIFESTS_DIR" || print_status "Manifests directory already exists: $K3S_MANIFESTS_DIR"
+    [[ ! -d "$K3S_IMAGES_DIR" ]] && mkdir -p "$K3S_IMAGES_DIR" && print_status "Created images directory: $K3S_IMAGES_DIR" || print_status "Images directory already exists: $K3S_IMAGES_DIR"
+}
+
+# Function to copy charts
+copy_charts() {
+    print_status "Copying Helm charts to $K3S_CHARTS_DIR..."
+    cp -v "$SCRIPT_DIR/charts"/*.tgz "$K3S_CHARTS_DIR/" && print_status "Successfully copied charts" || { print_error "Failed to copy charts"; exit 1; }
+}
+
+# Function to copy manifests
+copy_manifests() {
+    print_status "Copying manifest to $K3S_MANIFESTS_DIR..."
+    cp -v "$SCRIPT_DIR/manifests/trusted-compute.yaml" "$K3S_MANIFESTS_DIR/" && print_status "Successfully copied trusted-compute.yaml" || { print_error "Failed to copy trusted-compute.yaml"; exit 1; }
+}
+
+# Function to copy images
+copy_images() {
+    print_status "Copying container images to $K3S_IMAGES_DIR..."
+    cp -v "$SCRIPT_DIR/images"/*.tar "$K3S_IMAGES_DIR/" && print_status "Successfully copied container images" || { print_error "Failed to copy container images"; exit 1; }
+}
+
+# Function to check Ubuntu Secure Boot status
+check_secure_boot() {
+    print_status "Checking Secure Boot status..."
+    if mokutil --sb-state &>/dev/null; then
+        local sb_state=$(mokutil --sb-state | grep -i 'SecureBoot' | awk '{print $2}')
+        if [[ "$sb_state" == "enabled" ]]; then
+            print_status "Secure Boot is ENABLED."
+        else
+            print_warning "Secure Boot is DISABLED."
+            exit 1
+        fi
+    else
+        print_warning "mokutil not found or Secure Boot status cannot be determined."
+        exit 1
+    fi
+}
+
+# Function to copy containerd config template
+copy_containerd_config() {
+    print_status "Copying containerd config template..."
+    [[ ! -d "$CONTAINERD_ETC_DIR" ]] && mkdir -p "$CONTAINERD_ETC_DIR" && print_status "Created containerd etc directory: $CONTAINERD_ETC_DIR"
+    if [[ -f "$CONTAINERD_CONFIG_SRC" ]]; then
+        cp -v "$CONTAINERD_CONFIG_SRC" "$CONTAINERD_CONFIG_DEST" && print_status "Successfully copied containerd config template" || { print_error "Failed to copy containerd config template"; exit 1; }
+    else
+        print_error "containerd config template not found: $CONTAINERD_CONFIG_SRC"
```

Comment:
not sure I understand this else case seems $CONTAINERD_CONFIG_SRC will always be created using mkdir -p . when will this else be reached ?